src/common/entity/compute_state_display.ts

```diff
@@ -5,12 +5,90 @@ import { formatTime } from "../datetime/format_time";
 import { LocalizeFunc } from "../translations/localize";
 import { computeStateDomain } from "./compute_state_domain";
 import { UNKNOWN, UNAVAILABLE } from "../../data/entity";
+import { atLeastCachedVersion } from "../config/version";
 
 export const computeStateDisplay = (
   localize: LocalizeFunc,
   stateObj: HassEntity,
   language: string
 ): string => {
+  if (__BACKWARDS_COMPAT__ && !atLeastCachedVersion(0, 109)) {
```

Comment:
Can we move this to a separate function?

Reply:
```ts
  if (__BACKWARDS_COMPAT__ && !atLeastCachedVersion(0, 109)) {
	return legacyComputeStateDisplay(localize, stateObj, language);
  }
```

Reply:
Done.